File: exercism/c/collatz-conjecture/src/collatz_conjecture.c
#include "collatz_conjecture.h"

int steps(int start) {
  if (start <= 0)
    return ERROR_VALUE;
  int count = 0;
  while (start != 1) {
    start = start & 1 ? 3 * start + 1 : start >> 1;
    count++;
  }
  return count;
}

File: exercism/go/collatz-conjecture/collatz_conjecture.go
package collatzconjecture

import "errors"

func CollatzConjecture(n int) (steps int, e error) {
	if n < 1 {
		return -1, errors.New("n must be >= 1")
	}
	for n > 1 {
		if n&1 == 0 {
			n >>= 1
		} else {
			n = 3*n + 1
		}
		steps += 1
	}
	return
}
